src/components/views/auth/signUp/SignUp.spec.js

```diff
@@ -69,4 +70,20 @@ describe('SignUp Component', () => {
     form.simulate('submit');
     expect(registerUser).toHaveBeenCalledTimes(1);
   });
+  it('should call history prop function to redirect users to dashboard on successful authentication', () => {
+    const history = {
+      push: jest.fn(),
+    };
+    const location = {
+      search: 'localhost:8080/signup?token=abc',
+    };
+    mount(
+      <Provider store={store}>
+        <Router>
+          <SignUp location={location} history={history} />
+        </Router>
+      </Provider>,
+    );
+    expect(history.push).toHaveBeenCalledTimes(2);
```

Comment:
Any idea why `history.push` is called twice?

Reply:
Also, did you have to add the store provider?

Reply:
> Any idea why `history.push` is called twice?

It's called the first time our backend comes with the token in the URL, and the second time when redirecting

Reply:
The first suggestion I see in the article is to test with a mock store, I'm wondering why you didn't go for that since existing tests went the mock way.

I personally don't worry about the redux store when testing components so I always go for the _Testing component without Redux store_ suggestion

Reply:
Okay. I have removed the store provider, just passed the store as a prop to my component, what do you think?

Reply:
I think you should use a mock store

Reply:
Okkay

Reply:
That was definitely an oversight. Please let's switch to using a mock store to test our components as well